File: frontend/types/news.ts
export interface NewsItem {
  id: string;
  title: string;
  slug: string;
  excerpt: string;
  content: string;
  coverImage?: string;
  publishedAt: string;
  category?: string;
  author?: string;
}

export interface NewsResponse {
  news: NewsItem[];
  total: number;
  page: number;
  hasMore: boolean;
}

File: backend/internal/models/news.go
package models

import "time"

type News struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Slug        string    `json:"slug"`
	Excerpt     string    `json:"excerpt"`
	Content     string    `json:"content"`
	CoverImage  string    `json:"coverImage,omitempty"`
	PublishedAt time.Time `json:"publishedAt"`
	Category    string    `json:"category,omitempty"`
	Author      string    `json:"author,omitempty"`
}

type NewsResponse struct {
	News    []News `json:"news"`
	Total   int    `json:"total"`
	Page    int    `json:"page"`
	HasMore bool   `json:"hasMore"`
}
